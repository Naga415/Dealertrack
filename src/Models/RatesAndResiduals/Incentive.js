'use strict';
export default class Incentive {
  constructor() {
    this.id =null;
    this.name = "";
    this.expirationDate = null;
    this.offer = null;
    this.moneyFactor = null;
    this.apr = null;
    this.details = "";
    this.exclusions=[];
  }
}
