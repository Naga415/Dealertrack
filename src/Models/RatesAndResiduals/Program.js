'use strict';
export default class Program {
  constructor() {
    this.id = null;
    this.type = null;
    this.lender = null;
    this.rebates = null;
    this.rateMoneyFactor = null;
    this.term = null;
    this.acquisitionFee = null;
    this.residual = null;
    this.milesPerYear = null;
    this.payment = null;
    this.frontGross = null;
    this.backGross = null;
    this.expirationDate = null;
    this.maxAdvance = null;
    this.amountFinanced = null;
  }
}
