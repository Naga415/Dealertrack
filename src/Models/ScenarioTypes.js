export const FINANCE = "Finance";
export const LEASE = "Lease";
export const CASH = "Cash";
export const BALLOON = "Balloon";
