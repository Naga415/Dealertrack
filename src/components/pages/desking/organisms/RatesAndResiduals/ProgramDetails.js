'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Incentive from '../../../../../models/RatesAndResiduals/Incentive';
import Program from '../../../../../models/RatesAndResiduals/Program';
import Button from '@coxautokc/fusion-ui-components/lib/Button';
import DataTable from '@coxautokc/fusion-ui-components/lib/DataTable';
import { getCurrencyFormattedNumber } from '../../../../../utils/numberFormatter';

// eslint-disable-next-line react/prefer-stateless-function
class ProgramDetails extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  showRRModal = () => {
    this.props.onShowRRModal(true);
  }

  getTotalIncentivesApplied = () => {
    let total = 0;
    if (this.props.selectedIncentives) {
      this.props.selectedIncentives.forEach(function (incentive) {
        total += incentive.offer;
      }, this);
    }
    return getCurrencyFormattedNumber(total);
  }

  buildColumnHeaders = () => {
    return [
      {
        columnName: "type",
        displayName: "Type",
      },
      {
        columnName: 'lender',
        displayName: 'Lender'
      },
      {
        columnName: 'rebates',
        displayName: 'Rebates'
      },
      {
        columnName: 'rateMoneyFactor',
        displayName: 'Rate/MF'
      },
      {
        columnName: 'term',
        displayName: 'Term'
      },
      {
        columnName: 'acquisitionFee',
        displayName: 'Acquisition Fee'
      },
      {
        columnName: 'residual',
        displayName: 'Res %'
      },
      {
        columnName: 'maxAdvance',
        displayName: 'Max Adv'
      }
    ];
  }

  render() {
    return (
      <div id={this.props.htmlId} className="program-details">
        <DataTable
          htmlId="AppliedIncentivesTable"
          className="program-details__table"
          columnHeaders={this.buildColumnHeaders()}
          data={[this.props.selectedProgram]}
          columns={["type", "lender", "rebates", "rateMoneyFactor", "term", "acquisitionFee", "residual", "maxAdvance"]}
          displayFilter={false}
          displayPagination={false}
        />
        <div className="program-details__incentives">Applied Incentives&nbsp;&nbsp;&nbsp;{this.getTotalIncentivesApplied()}</div>
        <Button
          htmlId="ViewDetailsLink"
          className='program-details__modal_link'
          bsStyle="link"
          onClick={this.showRRModal}>View Details</Button>
      </div>
    );
  }
}

ProgramDetails.propTypes = {
  htmlId: PropTypes.string.isRequired,
  selectedProgram: PropTypes.instanceOf(Program),
  selectedIncentives: PropTypes.arrayOf(PropTypes.instanceOf(Incentive)),
  selectedLender: PropTypes.string,
  onShowRRModal: PropTypes.func
};

export default ProgramDetails;
