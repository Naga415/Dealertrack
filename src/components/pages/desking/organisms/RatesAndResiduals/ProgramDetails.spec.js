import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import ProgramDetails from './ProgramDetails';
import * as ScenarioTypes from '../../../../../models/ScenarioTypes';
import Program from '../../../../../models/RatesAndResiduals/Program';
import Incentive from '../../../../../models/RatesAndResiduals/Incentive';

describe('<ProgramDetails>', () => {
  const onShowRRModal = sinon.spy();
  const selectedProgram = new Program();
  selectedProgram.id = 0;
  selectedProgram.type = ScenarioTypes.FINANCE;
  selectedProgram.acquisitionFee = 456;
  selectedProgram.amountFinanced = 40000;
  selectedProgram.backGross = 456;
  selectedProgram.expirationDate = '11/23/2020';
  selectedProgram.frontGross = 234;
  selectedProgram.lender = 'acd';
  selectedProgram.maxAdvance = 643;
  selectedProgram.payment = 900;
  selectedProgram.rebates = 1500;
  selectedProgram.term = 24;
  selectedProgram.rateMoneyFactor = 6.9;
  const militaryIncentive = new Incentive();
  militaryIncentive.id = 1;
  militaryIncentive.name = "Military Incentive";
  militaryIncentive.apr = '5.4';
  militaryIncentive.expirationDate = '12/05/2018';
  militaryIncentive.details = "Work hard, play harder";
  militaryIncentive.exclusions = [2];
  militaryIncentive.offer = 1234;
  const gradIncentive = new Incentive();
  gradIncentive.id = 2;
  gradIncentive.name = "Recent Grad Incentive";
  gradIncentive.apr = '5.4';
  gradIncentive.expirationDate = '12/05/2017';
  gradIncentive.details = "Work hard, play harder";
  gradIncentive.exclusions = [1, 3];
  gradIncentive.offer = 1234;
  const props = {
    htmlId: "ProgramDetails",
    selectedProgram,
    selectedIncentives: [militaryIncentive, gradIncentive],
    selectedLender: "Ally",
    onShowRRModal
  };

  it('renders <ProgramDetails/>', () => {
    const component = renderer.create(<ProgramDetails {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('it displays aggreate of the applied incentives', () => {
    const expected = "Applied Incentives   $2,468";
    const component = mount(<ProgramDetails {...props} />);
    expect(component.find('.program-details__incentives').text()).toContain(expected);
  });

  it('it invokes onShow when View Details is clicked', () => {
    const component = mount(<ProgramDetails {...props} />);
    component.find('#ViewDetailsLink').simulate('click');
    expect(onShowRRModal.calledOnce).toEqual(true);
  });
});

