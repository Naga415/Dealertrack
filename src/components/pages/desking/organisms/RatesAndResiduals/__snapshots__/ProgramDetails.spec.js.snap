// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgramDetails> renders <ProgramDetails/> 1`] = `
<div
  className="program-details"
  id="ProgramDetails"
>
  <div
    className="data-table griddle program-details__table"
    id="AppliedIncentivesTable"
  >
    <div
      className={undefined}
      style={undefined}
    >
      <div
        className={undefined}
        style={undefined}
      />
      <table
        className="griddle-table"
        style={undefined}
      >
        <thead
          className="griddle-table-heading"
          style={undefined}
        >
          <tr>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Type
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Lender
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Rebates
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Rate/MF
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Term
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Acquisition Fee
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Res %
              </span>
            </th>
            <th
              className="griddle-table-heading-cell"
              onClick={[Function]}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span>
                Max Adv
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className="griddle-table-body"
          style={undefined}
        >
          <tr
            className="griddle-row"
            style={undefined}
          >
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              Finance
            </td>
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              acd
            </td>
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              1500
            </td>
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              6.9
            </td>
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              24
            </td>
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              456
            </td>
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            />
            <td
              className="griddle-cell"
              onClick={undefined}
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={
                Object {
                  "width": null,
                }
              }
            >
              643
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="program-details__incentives"
  >
    Applied Incentives   
    $2,468
  </div>
  <button
    className="program-details__modal_link btn btn-link"
    disabled={false}
    href={undefined}
    id="ViewDetailsLink"
    onClick={[Function]}
    type="button"
  >
    View Details
  </button>
</div>
`;
